### PR TITLE
[AscendNPU-IR][Expert] Support Sigmoid Op

### PR DESCRIPTION
**Pull request by dangyw18**

### CODE REVIEW

File: tilelang/language/customize_npuir.py [hunk 1]
```diff
@@ -544,7 +544,44 @@ def _tir_call_intrin(channel_nums, index_mode, src, *dsts: tir.PrimExpr):
...
+    
+    if dst is None:
+        dst = src
+    
+    if tmp is None:
+        raise ValueError("sigmoid operation requires a temporary buffer")
+    
+    neg_one = tir.const(-1.0, dtype)
+    mul_call = AscendBinaryOp("mul", src, neg_one, tmp).buildTirCall()
+    exp_call = AscendUnaryOp("exp", tmp, tmp).buildTirCall()  # 结果存回tmp
```
Comment (CeleNewYear) [inline tilelang/language/customize_npuir.py]: Please use English comments.

File: testing/npuir/test_sigmoid.py [hunk 2]
```diff
@@ -0,0 +1,91 @@
+import torch
```
Comment (CeleNewYear) [inline testing/npuir/test_sigmoid.py:1]: move the test case to testing/npuir/

**Comment by dangyw18** [comment]
support T.sigmoid Op
usage: T.sigmoid(src, dst, tmp)

**Review by baberrehman — DISMISSED** [review]
Please provide description of the feature.

File: tilelang/language/customize_npuir.py [hunk 3]
```diff
@@ -742,7 +742,44 @@ def _tir_call_intrin(channel_nums, index_mode, src, *dsts: tir.PrimExpr):
...
+    """
+    
+    shape = src.shape
+    dtype = src.dtype
+    
+    if dst is None:
+        dst = src
+    
+    if tmp is None:
+        raise ValueError("sigmoid operation requires a temporary buffer")
```
Comment (CeleNewYear) [inline tilelang/language/customize_npuir.py]: Forcing users to declare the tmp space is not reasonable. You can refer to the reduce OP for auxiliary generation.

**Review by CeleNewYear — CHANGES_REQUESTED** [review]

**Review by CeleNewYear — APPROVED** [review]
